server/src/main/java/facultad/trendz/service/PostService.java

```diff
@@ -64,35 +72,37 @@ public PostResponseDTO editPost(PostEditDTO postEdit, Long postId) {
 
             Post editedPost = postRepository.save(post1);
 
+            List<Comment> comments = editedPost.getComments();
+            comments.removeIf(Comment::isDeleted);
+
             return new PostResponseDTO(editedPost.getId(),
                     editedPost.getTitle(),
                     editedPost.getDescription(),
                     editedPost.getLink(),
                     editedPost.getDate(),
                     editedPost.getTopic().getId(),
-                    editedPost.getUser().getId());
+                    editedPost.getUser().getId(),
+                    commentListToDTO(comments),
+                    editedPost.getUser().getUsername());
         }).orElseThrow(PostNotFoundException::new);
     }
 
-    public PostGetDTO getPost(Long postId){
+    public PostResponseDTO getPost(Long postId){
         final Optional<Post> post= postRepository.findById(postId);
-        if (!post.isPresent()) throw new PostNotFoundException();
-
-        return new PostGetDTO(post.get().getTopic().getId(),
-                post.get().getTitle(),
-                post.get().getDescription(),
-                post.get().getLink(),
-                post.get().getDate(),
-                post.get().getUser().getId(),
-                post.get().getUser().getUsername()
-        );
-    }
-
-    public Long getPostAuthor(Long postId){
-        final Optional<Post> post= postRepository.findById(postId);
-        return post.map(post1 ->
-                post1.getUser().getId()
-        ).orElseThrow(PostNotFoundException::new);
+        return post.map(foundPost -> {
+            List<Comment> foundComments = foundPost.getComments();
+            foundComments.removeIf(Comment::isDeleted);
```

Comment:
@Agusdelavega Implementation is just fine, but better use streams! Also, please sort comments by created date so that they are shown first in frontend
```suggestion
            List<Comment> foundComments = foundPost.getComments()
            .stream()
            .filter(// filter by non deleted)
            .sort(// sort by created date)
            .collectors(// collect to convenient data type)
```